### ISSUE TITLE
don't depend on LANG=utf-8 being set for interpreting cabal files' encoding properly

### ISSUE THREAD

**Issue by sol** [NixOS/cabal2nix#333]
Hey!  This probably obvious to everybody with minimal Nix experience, but I lack that basic experience. So here is what I try:

```
$ nix-shell -p cabal2nix --pure --run 'cabal2nix .'
*** parsing cabal file: ././tinc.cabal: hGetContents: invalid argument (invalid byte sequence)
cabal2nix: nix-prefetch-zip: createProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
```

If I run this without `--pure` or install `cabal2nix` into my profile then it works, even though there is no `nix-prefetch-zip` on my `PATH`.  What additional package do I need to pass to `nix-env` to make this work?

**Comment by sol** [comment]
Indeed, the following works:

```
nix-shell -p cabal2nix --pure --run 'LC_ALL=en_US.UTF-8 cabal2nix .'
```

Note that `.cabal`-files are UTF-8 per specification.  So at this point I would label this a bug, either in `cabal2nix` or one of it's dependencies (`nix-prefetch-zip`?!).

**Comment by sol** [comment]
For completeness: https://www.haskell.org/cabal/users-guide/developing-packages.html#creating-a-package

> a Unicode UTF-8 text file containing a package description

**Comment by peti** [comment]
This is a bug in `readGenericPackageDescription` from `Cabal`, which depends on the system locale to interpret Cabal files as utf-8-encoded. As far as I can tell, this issue is fixed in the upcoming 2.2.x release.
